### PR TITLE
Fixed broken partial tags in Xml documentation 

### LINKED ISSUE #8322: XML Doc Comment tag completion is broken for partial tags
Source:

```C#
///<r
```
(or any string after the `<`)

Pressing Esc (if an existing completion list is open), then Ctrl+Space at the end will only show `see` and `seealso`.
Deleting the character, pressing Ctrl+Space at `<`, then retyping the character will correctly show every completion.

### PR DESCRIPTION
**Pull request by tvand7093**
This should resolve #8322. I'm new to the project so please let me know if something is wrong with this!

### CODE REVIEW

**Comment by Pilchie** [comment]
Thanks for the submission - tagging @dotnet/roslyn-ide for code review.

Also, @dotnet-bot test vsi please, test eta please.

File: src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs [hunk 1]
```diff
@@ -95,7 +95,8 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
                     items.AddRange(GetListHeaderItems(span));
                 }
 
-                if (token.Parent.Parent is DocumentationCommentTriviaSyntax)
+                if (token.Parent.Parent is DocumentationCommentTriviaSyntax ||
+                    (token.Parent.Parent.IsKind(SyntaxKind.XmlEmptyElement) && token.Parent.Parent.Parent is DocumentationCommentTriviaSyntax))
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs]: Just happened to notice that line 84 uses a single `&`.  Nothing to do with your change, but maybe if others have PR feedback that brings you back here you could fix that while you're here :)

**Comment by Pilchie** [comment]
:+1:

File: src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs (hunk 1 above)
Comment (tvand7093) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs]: I saw that actually, and just figured it was some voodoo that someone was doing so I left it alone. 😉

Comment (rchande) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs]: This isn't quite right. Consider:
```C#
///<summary>
/// <r$$
///</summary>
```

In that case, the token's great-grandparent (`token.Parent.Parent.Parent`) is an XmlElementSyntax.

Comment (tvand7093) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/XmlDocCommentCompletionProvider.cs]: I see... I'll take a look tonight and hopefully get this taken care of.

**Comment by tvand7093** [comment]
Sorry for all the noise on this PR. Still getting the hang of the repository... Noob question: how can I see why the CI build `windows_vsi_p3_open_prtest` failed?

**Comment by rchande** [comment]
No worries, @tvand7093  :smile:. Thanks for taking a look at this for us!

**Comment by rchande** [comment]
retest vsi

**Comment by rchande** [comment]
@dotnet-bot retest windows_vsi_p3_open_prtest please

**Comment by rchande** [comment]
:+1:

**Comment by Pilchie** [comment]
Was looking through old PRs and saw that this never got merged.  @tvand7093 do you want to bring it up to day, or do you want one of us to?

**Comment by tvand7093** [comment]
@Pilchie I can do it right now actually. I''ll let you know when it's done.

**Comment by tvand7093** [comment]
Alright @Pilchie, all caught up.

**Comment by Pilchie** [comment]
Thanks!  I'm going to re-test, since it's been a while, but assuming that goes well, we'll get this merged.

**Comment by Pilchie** [comment]
@dotnet-bot retest this please

**Comment by Pilchie** [comment]
@dotnet-bot test vsi please

**Comment by Pilchie** [comment]
Thanks again for the contribution, and sorry for the delay in merging it!

**Pull request merged**